client/client.go

```diff
@@ -144,6 +147,24 @@ func New(cfg Config) (Client, error) {
 	return c, nil
 }
 
+// KeepSync keeps synchronizing with etcd cluster about its member list,
+// so client could update its endpoints and catch up the change of the cluster.
+func KeepSync(ctx context.Context, c Client) error {
```

Comment:
The only reason http.Serve blocks is that you can consume the returned error. 

There is a ctx passed in which might cause the error. Actually I do not know why we want to return the error. The only thing I can think of is a canceled error that is triggered by the user and it does not need to be consumed.